xbmc/video/dialogs/GUIDialogVideoInfo.cpp

```diff
@@ -892,6 +892,38 @@ void CGUIDialogVideoInfo::OnGetArt()
   };
 }
 
+namespace
+{
+std::string GetEmbeddedArt(const std::string& fileNameAndPath, const std::string& artType)
+{
+  std::string embeddedArt;
+  if (CServiceBroker::GetSettingsComponent()->GetSettings()->GetBool(
+          CSettings::SETTING_MYVIDEOS_USETAGS))
+  {
+    if (URIUtils::HasExtension(fileNameAndPath, ".mkv") ||
+        URIUtils::HasExtension(fileNameAndPath, ".mp4") ||
+        URIUtils::HasExtension(fileNameAndPath, ".avi") ||
+        URIUtils::HasExtension(fileNameAndPath, ".m4v"))
+    {
```

Comment:
Looking at the call frequency of this function (once per preparing the list of available art for the selection dialog, I tend to say that this is nothing that really needs to be fine-tuned for performance.

I will go with `URIUtils::HasExtension(fileNameAndPath, ".mkv|.mp4|.avi|.m4v")` as this simplifies the code and is even a bit faster compared to what I coded originally, according to your numbers.

Reply:
But let me make this change in a followup-PR, please. I have a major refactoring in the pipeline, where this code will be moved to another source file anyway.

Reply:
@AlwinEsch std::set<> instead of vector?

Reply:
> std::set<> instead of vector?

Why? Can a set be faster iterated than a vector?

Reply:
> @AlwinEsch std::set<> instead of vector?

Hi @howie-f Then I think better to use `std::unordered_set<>`, think no need to sort something about.

> But let me make this change in a followup-PR, please. I have a major refactoring in the pipeline, where this code will be moved to another source file anyway.

Hi @ksooo, sorry written a bit bad, you can do in your request by the already present functions in Kodi, only to change to the the `|`.
Thought to ask what you think to have the new in future, not for now here.

Reply:
> Why? Can a set be faster iterated than a vector?

in std::set, which is sorted, the lookup operation is not an iteration over the elements. it may be not relevant with such small sets that we have here.

Reply:
The container holding the file extensions only will be interested pvr by one by the new HasExtension method, no lookup at all. So if we really want to performance optimize this function, we need a container that can be fast iterated, not random access.

Reply:
@AlwinEsch  the new function makes sense, but only if you not only introduce it, but also change exiting code to use it with the same PR.

Reply:
> @AlwinEsch the new function makes sense, but only if you not only introduce it, but also change exiting code to use it with the same PR.

Yes, without change existing code makes no sense.